specs/electra/beacon-chain.md

```diff
@@ -1341,24 +1340,16 @@ def apply_deposit(state: BeaconState,
         # Verify the deposit signature (proof of possession) which is not checked by the deposit contract
         if is_valid_deposit_signature(pubkey, withdrawal_credentials, amount, signature):
             add_validator_to_registry(state, pubkey, withdrawal_credentials, Gwei(0))  # [Modified in Electra:EIP7251]
-            # [New in Electra:EIP7251]
-            state.pending_deposits.append(PendingDeposit(
-                pubkey=pubkey,
-                withdrawal_credentials=withdrawal_credentials,
-                amount=amount,
-                signature=signature,
-                slot=GENESIS_SLOT,  # Use GENESIS_SLOT to distinguish from a pending deposit request
-            ))
-    else:
-        # Increase balance by deposit amount
-        # [Modified in Electra:EIP7251]
-        state.pending_deposits.append(PendingDeposit(
-            pubkey=pubkey,
-            withdrawal_credentials=withdrawal_credentials,
-            amount=amount,
-            signature=signature,
-            slot=GENESIS_SLOT  # Use GENESIS_SLOT to distinguish from a pending deposit request
-        ))
+
```

Comment:
consider:

new public key
invalid deposit signature

status quo: no pending deposit is made (correct behavior)
this PR: pending deposit is made (incorrect behavior as we didn't verify the proof-of-possession which is important for bls security)

Reply:
Oops yes. Missed that somehow. Thank you. Pushed a commit to fix this.

Also, since all tests passed, this implies we are missing some test coverage here.